File: frontend/src/types/default-account.ts
export interface DefaultAccount {
  id: string;
  setting_key: string;
  account_id: string | null;
  account_name: string | null;
  account_number: string | null;
  description: string | null;
}

export interface DefaultAccountUpdate {
  setting_key: string;
  account_id: string | null;
}

File: backend/crates/konto-api/src/dto/default_account.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Debug, Clone, Serialize, ToSchema)]
pub struct DefaultAccountResponse {
    pub id: String,
    pub setting_key: String,
    pub account_id: Option<String>,
    pub account_name: Option<String>,
    pub account_number: Option<String>,
    pub description: Option<String>,
}

#[derive(Debug, Deserialize, ToSchema)]
pub struct UpdateDefaultAccountsRequest {
    pub settings: Vec<DefaultAccountUpdate>,
}

#[derive(Debug, Deserialize, ToSchema)]
pub struct DefaultAccountUpdate {
    pub setting_key: String,
    pub account_id: Option<String>,
}
